### PR TITLE
tests: start teaching the fakestore about assertions

### PR DESCRIPTION
**Pull request by pedronis**

### CODE REVIEW

File: asserts/snap_asserts.go [hunk 1]
```diff
@@ -111,6 +112,21 @@ func assembleSnapDeclaration(assert assertionBase) (Assertion, error) {
 	}, nil
 }
 
+// SnapFileSHA3_384 computes the SHA3-384 digest of the given snap file.
+// It also returns its size.
+func SnapFileSHA3_384(snapPath string) (digest string, size uint64, err error) {
```
Comment (mvo5) [inline asserts/snap_asserts.go]: (nitpick) Maybe `SHA3_384Digest()` or something? As this will work on any file?

Comment (pedronis) [inline asserts/snap_asserts.go]: yes, but this is inside the asserts package (and snap_asserts.go to be precise), it's used to compute the snap-sha3-384 field basically of the snap-revision/snap-build, I would keep it named like that until it's here and we really have no generic use case

File: overlord/assertstate/assertmgr.go [hunk 2]
```diff
@@ -180,31 +176,13 @@ func doFetchSnapAssertions(t *state.Task, _ *tomb.Tomb) error {
 	}
 
 	// TODO: when we actually use this, snapPath might come from ss
-	// and switch likely to osutil.FileDigest and decide where/when
-	// to actually compute the hash
 	snapPath := snap.MinimalPlaceInfo(ss.Name(), ss.Revision()).MountFile()
 
-	snapf, err := snap.Open(snapPath)
+	sha3_384, _, err := asserts.SnapFileSHA3_384(snapPath)
```
Comment (mvo5) [inline overlord/assertstate/assertmgr.go]: Nice to see how the above simplifies the code here :)

File: asserts/snap_asserts.go (hunk 1 above)
Comment (pedronis) [inline asserts/snap_asserts.go]: we have a place if it really useful for other files (but still assertion related) which would be asserts/digest.go here

**Comment by mvo5** [comment]
Looks fine! Really cool to see that the fake store can handle this now :-D

**Pull request merged**